src/graph.cc

```diff
@@ -591,15 +594,18 @@ bool ImplicitDepLoader::LoadDepsFromLog(Edge* edge, string* err) {
     return false;
   }
 
-  vector<Node*>::iterator implicit_dep =
-      PreallocateSpace(edge, deps->node_count);
-  for (int i = 0; i < deps->node_count; ++i, ++implicit_dep) {
-    Node* node = deps->nodes[i];
+  ApplyLoadedDeps(edge, deps->nodes, deps->node_count);
+  return true;
+}
+
+void ImplicitDepLoader::ApplyLoadedDeps(Edge* edge, Node** nodes, int node_count) {
+  vector<Node*>::iterator implicit_dep = PreallocateSpace(edge, node_count);
```

Comment:
Only now being made aware of the PreallocateSpace function, I think it's an abomination.

Can you instead use std::vector::reserve()?

I'd also argue that you should eliminate that function entirely, but if you don't I certainly will.

Reply:
As much as I don't fancy _either_ PreallocateSpace _or_ vector::reserve(), this looks a bit too close to the hot path to fiddle with as part of a larger change. I'd much rather see such changes done separately.

Reply:
Sure.